### PR TITLE
Av 1050 barometer implementation

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
Add barometer implementation for the thread and sensor

### CODE REVIEW

File: MIDAS/src/hardware/Barometer.cpp [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+
+    float pressure = static_cast<float>(MS.getPressure() * 0.01 + 26.03);
+    float temperature = static_cast<float>(MS.getTemperature() * 0.01);
+
+    Barometer b;
+
+    b.pressure = pressure;
+    b.temperature = temperature;
+
+    return b;
```
Comment (redindelible) [inline MIDAS/src/hardware/Barometer.cpp:35]: Just use designated initializer syntax to make the return value in place, i.e. `return { .pressure = pressure, .temperature = temperature };`.

**Review by redindelible — APPROVED** [review]
Just one place the code could be improved slightly, otherwise it looks good.

File: MIDAS/src/systems.cpp [hunk 2]
```diff
@@ -20,8 +20,9 @@ DECLARE_THREAD(data_logger, RocketSystems* arg) {
 
 DECLARE_THREAD(barometer, RocketSystems* arg) {
     while (true) {
+        arg->rocket_state.barometer.update(arg->sensors.barometer.read());
+
         THREAD_SLEEP(16);
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:25]: max sampling rate again

**Review by redindelible — DISMISSED** [review]

File: MIDAS/src/hardware/Barometer.cpp [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
+*/
 Barometer BarometerSensor::read() {
-    // read from aforementioned global instance of sensor
-    return Barometer();
+    MS.read(12);
+
+    float pressure = static_cast<float>(MS.getPressure() * 0.01 + 26.03);
+    float temperature = static_cast<float>(MS.getTemperature() * 0.01);
+
+    return { .pressure = pressure, .temperature = temperature };
```
Comment (aidancostello) [inline MIDAS/src/hardware/Barometer.cpp:30]: ok at this point idk if this is just a me problem but again this doesn't build in platformio

**Review by aidancostello — CHANGES_REQUESTED** [review]

**Review by aidancostello — APPROVED** [review]

File: MIDAS/src/systems.cpp [hunk 4]
```diff
@@ -20,8 +20,9 @@ DECLARE_THREAD(data_logger, RocketSystems* arg) {
 
 DECLARE_THREAD(barometer, RocketSystems* arg) {
     while (true) {
-        THREAD_SLEEP(16);
-        //Serial.println("BARO");
+        arg->rocket_state.barometer.update(arg->sensors.barometer.read());
+
+        THREAD_SLEEP(1);
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:25]: is this too short is anything else going to actually happen with this short of a delay @redindelible @NPIPHI

Comment (redindelible) [reply MIDAS/src/systems.cpp:25]: FreeRTOS's scheduler is advanced enough that all threads should run fine. We will need to tune this once we've got the boards though.

Comment (aidancostello) [reply MIDAS/src/systems.cpp:25]: got it, will merge

**Pull request merged**